tests/test_prometheus_alerts_configmap.py

```diff
@@ -1,50 +1,102 @@
+from . import supported_k8s_versions
+import re
+import pytest
 
 
-def process_record(rule):
-    """Process a record rule.
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestPrometheusAlertConfigmap:
+    show_only = ["charts/prometheus/templates/prometheus-alerts-configmap.yaml"]
 
-    https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/
-    """
-    assert isinstance(rule.get("expr"), str)
-    assert isinstance(rule.get("record"), str)
+    @staticmethod
+    def process_record(rule):
+        """Process a record rule.
 
+        https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/
+        """
+        assert isinstance(rule.get("expr"), str)
+        assert isinstance(rule.get("record"), str)
 
-def process_alert(rule):
-    """Process an alert rule.
+    @staticmethod
+    def process_alert(rule):
+        """Process an alert rule.
 
-    https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/
-    """
-    assert isinstance(rule.get("alert"), str)
-    assert isinstance(rule.get("expr"), str)
-    if rule["alert"] != "Watchdog":
-        assert isinstance(rule.get("for"), str)
-    assert isinstance(rule.get("labels"), dict)
-    assert isinstance(rule.get("annotations"), dict)
+        https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/
+        """
+        assert isinstance(rule.get("alert"), str)
+        assert isinstance(rule.get("expr"), str)
+        if rule["alert"] != "Watchdog":
+            assert isinstance(rule.get("for"), str)
+        assert isinstance(rule.get("labels"), dict)
+        assert isinstance(rule.get("annotations"), dict)
 
+    def test_prometheus_alerts_configmap(self, kube_version):
+        """Validate the prometheus alerts configmap and its embedded data."""
+        docs = render_chart(
+            kube_version=kube_version,
+            show_only=self.show_only,
+        )
 
-def test_prometheus_alerts_configmap():
-    """Validate the prometheus alerts configmap and its embedded data."""
-    docs = render_chart(
-        show_only=["charts/prometheus/templates/prometheus-alerts-configmap.yaml"],
-    )
+        assert len(docs) == 1
 
-    assert len(docs) == 1
+        doc = docs[0]
 
-    doc = docs[0]
+        assert doc["kind"] == "ConfigMap"
+        assert doc["apiVersion"] == "v1"
+        assert doc["metadata"]["name"] == "RELEASE-NAME-prometheus-alerts"
 
-    assert doc["kind"] == "ConfigMap"
-    assert doc["apiVersion"] == "v1"
-    assert doc["metadata"]["name"] == "RELEASE-NAME-prometheus-alerts"
+        # Validate the contents of an embedded yaml doc
+        groups = yaml.safe_load(doc["data"]["alerts"])["groups"]
+        for group in groups:
+            assert isinstance(group.get("name"), str)
+            assert isinstance(group.get("rules"), list)
+            for rule in group["rules"]:
+                if rule.get("alert"):
```

Comment:
I have a personal preference for `if "alert" in rule`. but it's just a nitpick

Reply:
it might be nice to have a comment or docstring showing what a rule actually looks like

Reply:
Already there: https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/